jax/_src/device_array.py

```diff
@@ -60,12 +60,6 @@ def make_device_array(
   This is to be used only within JAX. It will return either a PythonDeviceArray
   or a C++ equivalent implementation.
   """
-  warnings.warn(
-        'DeviceArray has been deprecated. '
-        'Please use `jax.Array`. See '
-        'https://jax.readthedocs.io/en/latest/jax_array_migration.html on how '
-        'to migrate to `jax.Array`.', DeprecationWarning)
```

Comment:
I think having the warning print only once would be good.
From memory, the warnings.warn(..., stacklevel=1) help archive that.